### ISSUE TITLE
calling `self.log(..., on_step=True)` in optimizer_step fails

### ISSUE THREAD

**Issue by quinor** [PyTorchLightning/pytorch-lightning#4467]
## 🐛 Bug

calling `self.log(..., on_step=True)` in optimizer_step fails.

### Expected behavior

It should get treated same way if I were to `.log()` in training_step

### Additional context

I believe it's connected to #4439. Tried debugging and hotfixing it but the whole .log() behavior (controlled by `self._current_fx_name`) seems not to be implemented too well.

**Comment by tchaton** [comment]
Hey @quinor,

You are right, we are currently refactoring logging to enable `self.log` everywhere.

I am curious, what do you intent to log in `optimizer_step`.

Best regards,
T.C

**Comment by quinor** [comment]
hey @tchaton,

As for your question: I'm trying to log the LR. To be precise,

```python
self.log("train/weight_lr", optimizer.param_groups[0]["lr"], on_step=True)
```

**Comment by rohitgr7** [comment]
@quinor there is LearningRateMonitor you can use to log learning rates.

**Comment by edenlightning** [comment]
Closing issue for now. @quinor feel free to reopen if you have more questions!

**Comment by vincentzlt** [comment]
have the same issue. Want to log batch accuracy at optimizer_step after gradient accumulation. Seems no other better place to log this..
